UPGRADING.md

```diff
@@ -31,6 +31,16 @@ amino to proto3 encoding and breaking changes to the header.
   [Cosmos-SDK](https://github.com/cosmos/cosmos-sdk).
   (https://github.com/cosmos/cosmos-sdk/blob/master/crypto/types/multisig/multisignature.go)
 
+### Protobuf
+
+With this release we are happy to announce the full protobuf migration of the Tendermint repo. This consists of changes that you may need to be aware of:
+
+- All proto files have been moved under one directory, `/proto`. This is in line with the recommended file layout by [buf](https://buf.build), you can read more about it [here](https://buf.build/docs/lint-checkers#file_layout)
+- We use the generated protobuf types for only on disk and over the wire serialization. This means that these changes should not effect you as user of Tendermint.
+- A few notable changes in the abci:
+  - In `ValidatorUpdates` the public key type has been migrated to a protobuf `oneof` type. Since Tendermint only supports ed25519 validator keys this is the only available key in the oneof.
+  - AppStateBytes is protobuf encoded instead of amino encoded <!-- TODO: see if this was even amino encoded before-->
```

Comment:
Have been reading this spec and AppStateBytes is mentioned as being amino encoded. Was this actually the case tho? 

This means that applications built in any lang would need amino to work??? right

Reply:
amino-encoded JSON (https://github.com/tendermint/tendermint/blob/95d2d136cdcff2a3bb29d4e4271d757e4f88867c/consensus/replay.go#L311)

> This means that applications built in any lang would need amino to work??? right

no, they will need a JSON parser

Reply:
hmm the sdk is talking about supporting both proto and amino encoded json for genesis, but based off this its not possible..

Reply:
note: Tendermint only initially sends json-encoded app_state from genesis.json to the app. at no point after that Tendermint uses AppStateBytes.

Reply:
ah okay, thank you for that